### DISCUSSION TITLE
API for self hosted runners

### DISCUSSION THREAD

**Discussion by buremba** [community/community#26436] [Actions]
<p>We would like to build a UI for the concurrent jobs running in our self hosted runner, while Github API has an API for actions, it’s rate-limited so polling the state every x seconds and fetching the logs exhaust our limits very quickly. Is there any way to talk the runner directly via an API?</p>

**Comment by mscoutermarsh** [comment, marked as answer]
<p>Hi <a class="mention" href="/buremba">@buremba</a>,</p>
<p>We currently only have what’s available in the<a href="https://developer.github.com/v3/actions/self-hosted-runners" rel="nofollow noopener">Actions API</a>. There are no endpoints for connecting directly to the runner. </p>
